swift-service/src/main/java/com/facebook/swift/service/ThriftMethodProcessor.java

```diff
@@ -220,9 +222,11 @@ public void onFailure(Throwable t)
                         } else {
                             contextChain.undeclaredUserException(t);
```

Comment:
This calls a user callback that can set headers. We have some predefined callback class that does.